UnityProject/Assets/Scripts/NPC/AI/MobMeleeAction.cs

```diff
@@ -92,99 +92,102 @@ protected virtual bool CheckForTargetAction()
 			}
 
 			var dirToTarget = (followTarget.position - transform.position).normalized;
-			RaycastHit2D hitInfo = Physics2D.Linecast(
-				transform.position + dirToTarget,
-				followTarget.position,
-				checkMask);
+			var hitInfo = MatrixManager.Linecast(
+				transform.position + dirToTarget, LayerTypeSelection.Windows,checkMask,
+				followTarget.position);
 
-			if (hitInfo.collider == null)
+			if (hitInfo.ItHit == false)
 			{
 				return false;
 			}
 
-			if (!(Vector3.Distance(transform.position, hitInfo.point) < 1.5f))
+			if (!(Vector3.Distance(transform.position, hitInfo.TileHitWorld) < 1.5f))
```

Comment:
Codacy found an issue: [Use the opposite operator ('>=') instead.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6319358)